### PR TITLE
Add CommandLinePathFactory to CommandEnvironment

### PR DESCRIPTION
**Pull request by Yannic**
Progress on #15856

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java [hunk 1]
```diff
@@ -273,6 +274,23 @@ public void exit(AbruptExitException exception) {
     }
     this.buildResultListener = new BuildResultListener();
     this.eventBus.register(this.buildResultListener);
+
+    ImmutableMap.Builder<String, Path> wellKnownRoots = ImmutableMap.builder();
+    // This is necessary because some tests don't have a workspace set.
+    putIfValueNotNull(wellKnownRoots, "workspace", directories.getWorkspace());
```
Comment (tjgq) [inline src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java:280]: The helper isn't used anywhere else, so just inline the logic here?

Comment (Yannic) [reply src/main/java/com/google/devtools/build/lib/runtime/CommandEnvironment.java]: Moved all logic into a static factory function for `CommandLinePathFactory`.

**Comment by Yannic** [comment]
PTAL

**Comment by Yannic** [comment]
@bazel-io flag

**Comment by ckolli5** [comment]
@bazel-io fork 5.3.0

**Comment by Yannic** [comment]
@tjgq ping?

**Pull request closed without merging**